### PR TITLE
Ensure the config directory, user and deal with a missing domain

### PR DESCRIPTION
**Pull request by ekohl**
This provides auto requires, for example when you want to manage certificates in the config directory.

### CODE REVIEW

**Comment by ehelms** [comment]
:+1: makes sense to me

File: manifests/config.pp [hunk 1]
```diff
@@ -40,10 +40,21 @@
...
+
+  group { $foreman_proxy::group:
+    system => true,
+  }
+
+  # Provided by packaging by provides autorequire for files
+  file { $foreman_proxy::config_dir:
+    ensure => directory,
+    owner  => 'root',
+    group  => 0,
```
Comment (evgeni) [inline manifests/config.pp:57]: any reason not to write `group => 'root'` here?

File: manifests/config.pp [hunk 2]
```diff
@@ -40,10 +40,21 @@
...
-    require => Class['foreman_proxy::install'],
-    notify  => Class['foreman_proxy::service'],
+    system  => true,
+  }
+
+  group { $foreman_proxy::group:
+    system => true,
+  }
+
+  # Provided by packaging by provides autorequire for files
```
Comment (evgeni) [inline manifests/config.pp:53]:
```suggestion
  # Provided by packaging, defined here to allow autorequire for files
```
Is that what you meant?

File: spec/classes/foreman_proxy__spec.rb [hunk 3]
```diff
@@ -70,15 +70,15 @@
...
-            :groups  => ['puppet'],
-            :home    => "#{home_dir}",
-            :require => 'Class[Foreman_proxy::Install]',
-            :notify  => 'Class[Foreman_proxy::Service]',
-          })
+          should contain_user("#{proxy_user_name}")
+            .with_ensure('present')
+            .with_shell(shell)
+            .with_comment('Foreman Proxy daemon user')
+            .with_group(proxy_user_name)
```
Comment (evgeni) [inline spec/classes/foreman_proxy__spec.rb:77]:
```suggestion
            .with_gid(proxy_user_name)
```

File: manifests/config.pp (hunk 1 above)
Comment (ekohl) [reply manifests/config.pp:57]: Technically we still mark it compatible with FreeBSD and writing `0` is shorter than `BSD ? 'wheel' : 'root'`. I copied it from elsewhere in this file.

File: manifests/config.pp (hunk 2 above)
Comment (ekohl) [reply manifests/config.pp:53]: Yes I did.

File: manifests/config.pp (hunk 1 above)
Comment (evgeni) [reply manifests/config.pp:57]: "ooof", okay

**Pull request merged**